src/execution.js

```diff
@@ -179,7 +179,8 @@ const metricToDefinition = metric => ({ element: get(metric, 'objectUri')});
 export const mdToExecutionConfiguration = (mdObj) => {
     const { measures, categories, filters } = mdObj;
     const factMetrics = map(filter(measures, m => m.type === 'fact'), generatedMetricDefinition);
-    const metrics = map(filter(measures, m => m.type === 'metric'), metricToDefinition);
+    const metricsWithFilters = map(filter(measures, m => m.type === 'metric' && notEmpty(m.metricAttributeFilters)), generatedMetricDefinition);
+    const metrics = map(filter(measures, m => m.type === 'metric' && isEmpty(m.metricAttributeFilters)), metricToDefinition);
```

Comment:
This would preserve the order of metrics:

```js
const metrics = map(filter(measures, m =>  m.type === 'metric'), metric => {
      if (isEmpty(metric.metricAttributeFilters)) {
          return generatedMetricDefinition(metric);
      }

      return metricToDefinition(metric);
});
```

```js
metrics.forEach({element, definition}) => {
    columns.push(element);

    if (definition) {
        definitions.push(definition);
    }
});
```